### PR TITLE
DAO8

### PR DESCRIPTION
**Pull request by yurwing**
added Controllers and .jsp for car, driver and manufacturer

### CODE REVIEW

File: src/main/webapp/WEB-INF/views/car/all.jsp [hunk 1]
```diff
@@ -0,0 +1,41 @@
+<%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
+<html>
+<head>
+    <title>All cars</title>
+</head>
+<body>
+<h1>All cars page</h1>
+
```
Comment (IgorBochkor) [inline src/main/webapp/WEB-INF/views/car/all.jsp:9]: delete empty line

File: src/main/webapp/WEB-INF/views/driver/all.jsp [hunk 2]
```diff
@@ -0,0 +1,37 @@
+<%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
+<html>
+<head>
+    <title>All drivers</title>
+</head>
+<body>
+<h1>All drivers page</h1>
+
```
Comment (IgorBochkor) [inline src/main/webapp/WEB-INF/views/driver/all.jsp:9]: delete empty line

File: src/main/webapp/WEB-INF/views/driver/create.jsp [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
+<html>
+<head>
+    <title>Title</title>
```
Comment (IgorBochkor) [inline src/main/webapp/WEB-INF/views/driver/create.jsp]: write correct title name

File: src/main/webapp/web.xml [hunk 4]
```diff
@@ -9,6 +9,98 @@
...
+
+    <servlet>
+        <servlet-name>addDriver</servlet-name>
+        <servlet-class>core.basesyntax.controllers.driver.CreateDriverController</servlet-class>
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>addDriver</servlet-name>
+        <url-pattern>/drivers/create</url-pattern>
+    </servlet-mapping>
+
```
Comment (IgorBochkor) [inline src/main/webapp/web.xml:50]: delete empty line

File: src/main/webapp/WEB-INF/views/manufacturer/create.jsp [hunk 5]
```diff
@@ -0,0 +1,18 @@
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
+<html>
+<head>
+    <title>Title</title>
```
Comment (IgorBochkor) [inline src/main/webapp/WEB-INF/views/manufacturer/create.jsp]: write correct title name

File: src/main/webapp/WEB-INF/views/car/create.jsp [hunk 6]
```diff
@@ -0,0 +1,19 @@
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
+<html>
+<head>
+    <title>Title</title>
```
Comment (IgorBochkor) [inline src/main/webapp/WEB-INF/views/car/create.jsp]: write correct title name

File: src/main/webapp/WEB-INF/views/car/addDriverToCar.jsp [hunk 7]
```diff
@@ -0,0 +1,18 @@
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
+<html>
+<head>
+    <title>Title</title>
```
Comment (IgorBochkor) [inline src/main/webapp/WEB-INF/views/car/addDriverToCar.jsp]: same

**Review by IgorBochkor — CHANGES_REQUESTED** [review]

**Review by IgorBochkor — APPROVED** [review]

File: src/main/java/core/basesyntax/controllers/car/AddDriverToCarController.java [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+        String carId = req.getParameter("carId");
+        if (driverId.isEmpty() || carId.isEmpty()) {
+            req.setAttribute("message", "ID must contains only digits.");
+            req.getRequestDispatcher("/WEB-INF/views/car/create.jsp").forward(req, resp);
+        }
+        if (!driverId.chars().allMatch(Character::isDigit)
+                || !carId.chars().allMatch(Character::isDigit)) {
+            req.setAttribute("message", "ID must contains only digits.");
+            req.getRequestDispatcher("/WEB-INF/views/car/create.jsp").forward(req, resp);
+        }
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/controllers/car/AddDriverToCarController.java]: let's remove this please

File: src/main/webapp/WEB-INF/views/car/addDriverToCar.jsp [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+    <title>Add driver to car</title>
+</head>
+<body>
+<h1>Enter data for adding driver to car</h1>
+
+<h2 style="color:red">${message}</h2>
+
+<form method="post" action="${pageContext.request.contextPath}/cars/add-driver">
+    Enter driver id: <input type="text" name="driverId">
+    Enter car id: <input type="text" name="carId">
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/views/car/addDriverToCar.jsp]:
```suggestion
    Enter driver id: <input type="number" name="driverId" required>
    Enter car id: <input type="number" name="carId" required>
```

File: src/main/java/core/basesyntax/controllers/car/CreateCarController.java [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+import core.basesyntax.service.CarService;
+import core.basesyntax.service.ManufacturerService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class CreateCarController extends HttpServlet {
+    private static Injector injector = Injector.getInstance("core.basesyntax");
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/controllers/car/CreateCarController.java]:
```suggestion
    private static final Injector injector = Injector.getInstance("core.basesyntax");
```
check that it is static and final in all your controllers

File: src/main/java/core/basesyntax/controllers/car/CreateCarController.java [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+        String model = req.getParameter("model");
+        String manufacturerId = req.getParameter("manufacturerId");
+
+        if (model.isEmpty() || manufacturerId.isEmpty()) {
+            req.setAttribute("message", "Input data cannot be empty.");
+            req.getRequestDispatcher("/WEB-INF/views/car/create.jsp").forward(req, resp);
+        } else if (!manufacturerId.chars().allMatch(Character::isDigit)) {
+            req.setAttribute("message", "Input data for manufacturer id must contain only digits.");
+            req.getRequestDispatcher("/WEB-INF/views/car/create.jsp").forward(req, resp);
+        } else {
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/controllers/car/CreateCarController.java]: let's remove this

File: src/main/java/core/basesyntax/controllers/driver/CreateDriverController.java [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String name = req.getParameter("name");
+        String licenceNumber = req.getParameter("licenceNumber");
+
+        if (name.isEmpty() || licenceNumber.isEmpty()) {
+            req.setAttribute("message", "Input data cannot be empty.");
+            req.getRequestDispatcher("/WEB-INF/views/driver/create.jsp").forward(req, resp);
+        } else {
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/controllers/driver/CreateDriverController.java]: same

File: src/main/java/core/basesyntax/controllers/manufacturer/CreateManufacturerController.java [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String name = req.getParameter("name");
+        String country = req.getParameter("country");
+
+        if (name.isEmpty() || country.isEmpty()) {
+            req.setAttribute("message", "Input data cannot be empty.");
+            req.getRequestDispatcher("/WEB-INF/views/manufacturer/create.jsp").forward(req, resp);
+        } else {
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/controllers/manufacturer/CreateManufacturerController.java]: same

File: src/main/java/core/basesyntax/dao/CarDaoJdbcImpl.java [hunk 14]
```diff
@@ -79,6 +79,7 @@ public Car create(Car car) {
         for (Car car : cars) {
             car.setDrivers(getDrivers(car.getId()));
         }
+        cars.sort((i1, i2) -> (int) (i1.getId() - i2.getId()));
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/dao/CarDaoJdbcImpl.java]: let's do this on the service level

File: src/main/webapp/web.xml [hunk 15]
```diff
@@ -9,6 +9,97 @@
...
+        <url-pattern>/</url-pattern>
     </servlet-mapping>
+
+    <servlet>
+        <servlet-name>allDrivers</servlet-name>
+        <servlet-class>core.basesyntax.controllers.driver.GetAllDriversController</servlet-class>
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>allDrivers</servlet-name>
+        <url-pattern>/drivers/all</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/drivers/</url-pattern>
```

File: src/main/webapp/web.xml [hunk 16]
```diff
@@ -9,6 +9,97 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>allCars</servlet-name>
+        <servlet-class>core.basesyntax.controllers.car.GetAllCarsController</servlet-class>
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>allCars</servlet-name>
+        <url-pattern>/cars/all</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]: same

File: src/main/webapp/web.xml [hunk 17]
```diff
@@ -9,6 +9,97 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>allManufacturer</servlet-name>
+        <servlet-class>core.basesyntax.controllers.manufacturer.GetAllManufacturerController</servlet-class>
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>allManufacturer</servlet-name>
+            <url-pattern>/manufacturers/all</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]: same

File: src/main/webapp/web.xml [hunk 18]
```diff
@@ -9,6 +9,97 @@
...
+        <url-pattern>/cars/create</url-pattern>
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>addDriverToCar</servlet-name>
+        <servlet-class>core.basesyntax.controllers.car.AddDriverToCarController</servlet-class>
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>addDriverToCar</servlet-name>
+        <url-pattern>/cars/add-driver</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/cars/drivers/add</url-pattern>
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/webapp/WEB-INF/views/car/addDriverToCar.jsp [hunk 19]
```diff
@@ -0,0 +1,18 @@
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
+<html>
+<head>
+    <title>Add driver to car</title>
+</head>
+<body>
+<h1>Enter data for adding driver to car</h1>
+
+<h2 style="color:red">${message}</h2>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/views/car/addDriverToCar.jsp]: if we don't need this - we can remove it
same for other JSPs

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**